frontend/src/components/PostList.tsx

```diff
@@ -0,0 +1,179 @@
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import postsService from "@/services/postsService";
+import { MoreHorizontal } from "lucide-react";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import PostContentModal from "./PostContentModal";
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "./ui/alert-dialog";
+import { Button } from "./ui/button";
+import { Dialog } from "./ui/dialog";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuGroup,
+  DropdownMenuItem,
+  DropdownMenuLabel,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+} from "./ui/dropdown-menu";
```

Comment:
It would be nice to configure absolute imports in the future so that instead of importing from "./ui" or "../ui" you will import from "ui/" or "@ui" or "~src/ui". Such an approach usually help in case you move thew file somewhere - import won't be changed